lib/promscrape/targetstatus.go

```diff
@@ -104,10 +105,53 @@ func (tsm *targetStatusMap) Reset() {
 
 func (tsm *targetStatusMap) registerJobNames(jobNames []string) {
 	tsm.mu.Lock()
+	tsm.registerJobsMetrics(jobNames)
 	tsm.jobNames = append(tsm.jobNames[:0], jobNames...)
 	tsm.mu.Unlock()
 }
 
+// registerJobsMetrics registers metrics for new jobs and unregisterMetric metrics for removed jobs
+func (tsm *targetStatusMap) registerJobsMetrics(jobNames []string) {
```

Comment:
Could you update var name here `preName` -> `prevName`? 

This is more consistent with other variables names used for similar purposes.